torch/linalg/__init__.py

```diff
@@ -1419,69 +1481,71 @@
 
     Non-default rcond example
     >>> rcond = 0.5
-    >>> a = torch.randn(3, 3)
-    >>> torch.linalg.pinv(a)
+    >>> A = torch.randn(3, 3)
+    >>> torch.linalg.pinv(A)
     tensor([[ 0.2971, -0.4280, -2.0111],
             [-0.0090,  0.6426, -0.1116],
             [-0.7832, -0.2465,  1.0994]])
-    >>> torch.linalg.pinv(a, rcond)
+    >>> torch.linalg.pinv(A, rcond)
     tensor([[-0.2672, -0.2351, -0.0539],
             [-0.0211,  0.6467, -0.0698],
             [-0.4400, -0.3638, -0.0910]])
 
     Matrix-wise rcond example
-    >>> a = torch.randn(5, 6, 2, 3, 3)
+    >>> A = torch.randn(5, 6, 2, 3, 3)
     >>> rcond = torch.rand(2)  # different rcond values for each matrix in a[:, :, 0] and a[:, :, 1]
-    >>> torch.linalg.pinv(a, rcond)
+    >>> torch.linalg.pinv(A, rcond)
     >>> rcond = torch.randn(5, 6, 2) # different rcond value for each matrix in 'a'
-    >>> torch.linalg.pinv(a, rcond)
+    >>> torch.linalg.pinv(A, rcond)
+
+.. _defined algebraically:
+    https://en.wikipedia.org/wiki/Moore%E2%80%93Penrose_inverse#Existence_and_uniqueness
+.. _through the SVD:
+    https://en.wikipedia.org/wiki/Moore%E2%80%93Penrose_inverse#Singular_value_decomposition_(SVD)
 """)
 
 solve = _add_docstr(_linalg.linalg_solve, r"""
-linalg.solve(input, other, *, out=None) -> Tensor
-
-Computes the solution of a square system  of linear equations with unique solution. Supports batched inputs.
-
-This method accepts both matrices and vectors as the right-hand side of the linear equation.
-We will denote by `*` zero or more batch dimensions.
+linalg.solve(A, B, *, out=None) -> Tensor
 
-- If :attr:`input`, :attr:`other` have shapes `(*, n, n)`, `(*, n)`, for every matrix :math:`A`
-  in :attr:`input` and vector :math:`b` in :attr:`other`, this function returns a vector :math:`x` such that
+Computes the solution of a square system of linear equations with unique solution.
```

Comment:
"with unique solution" -> "with a unique solution"

"solution of" -> "solution to"

Reply:
What happens if there's not a unique solution? Is the behavior different if there's no solution vs many solutions?

Reply:
Changed.

There's one solution iff A is invertible. If it's not, it throws an error, as documnted in `Raises:`.

Reply:
Also, added the point that this system has unique solution iff A is invertible.